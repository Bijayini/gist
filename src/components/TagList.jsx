import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';
import BodyCopy from '../ui-components/BodyCopy';

const TagList = ({ tags }) => {
  return (
    <Fragment>
      <div className="tag-container">
        {tags.map(tag => (
          <div className="tag" key={tag}>
            <BodyCopy variant="regular" xs="4" xm="4" xl="4" color="white">
              {tag}
            </BodyCopy>
          </div>
        ))}
      </div>
      <style jsx="true">
        {`
          .tag-container {
            display: flex;
          }
          .tag {
            background: ${theme.primaryColor};
            border-radius: 5px;
            margin-right: 5px;
            padding: 1px 5px;
          }
        `}
      </style>
    </Fragment>
  );
};

TagList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default TagList;
