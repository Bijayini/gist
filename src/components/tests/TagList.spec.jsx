import React from 'react';
import { shallow } from 'enzyme';

import TagList from '../TagList';

describe('<TagList />', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      tags: ['some tag', 'some tag2'],
    };
    component = shallow(<TagList {...props} />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
