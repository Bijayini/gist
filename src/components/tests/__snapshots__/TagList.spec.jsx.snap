// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagList /> should match with snapshot 1`] = `
"<Fragment>
  <div className=\\"jsx-3617528546 tag-container\\">
    <div className=\\"jsx-3617528546 tag\\">
      <BodyCopy variant=\\"regular\\" xs=\\"4\\" xm=\\"4\\" xl=\\"4\\" color=\\"white\\" className=\\"\\" uppercase={false} styles={false}>
        some tag
      </BodyCopy>
    </div>
    <div className=\\"jsx-3617528546 tag\\">
      <BodyCopy variant=\\"regular\\" xs=\\"4\\" xm=\\"4\\" xl=\\"4\\" color=\\"white\\" className=\\"\\" uppercase={false} styles={false}>
        some tag2
      </BodyCopy>
    </div>
  </div>
  <JSXStyle id=\\"2952745033\\" dynamic={{...}}>
    .tag-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.tag.__jsx-style-dynamic-selector{background:#cc1c5d;border-radius:5px;margin-right:5px;padding:1px 5px;}
  </JSXStyle>
</Fragment>"
`;
